src/model/CoilSystemCoolingWater.cpp

```diff
@@ -335,17 +323,20 @@ namespace model {
     OS_ASSERT(ok);
   }
 
-  CoilSystemCoolingWater::CoilSystemCoolingWater(const Model& model, const WaterToAirComponent& coolingCoil)
+  CoilSystemCoolingWater::CoilSystemCoolingWater(const Model& model, const HVACComponent& coolingCoil)
     : StraightComponent(CoilSystemCoolingWater::iddObjectType(), model) {
     OS_ASSERT(getImpl<detail::CoilSystemCoolingWater_Impl>());
 
     bool ok = true;
+    ok = setCoolingCoil(coolingCoil);
+    if (!ok) {
+      remove();
+      LOG_AND_THROW("Unable to set " << briefDescription() << "'s Cooling Coil " << coolingCoil.briefDescription() << ".");
+    }
```

Comment:
This is how you handle the explicit ctor being passed the wrong object, which can definitely happen.